addOns/ascanrulesAlpha/src/test/java/org/zaproxy/zap/extension/ascanrulesAlpha/ForbiddenBypassScanRuleUnitTest.java

```diff
@@ -74,22 +89,22 @@ void shouldNotAlertIfAllRequestsAreForbidden() throws Exception {
         rule.scan();
         // Then
         assertThat(alertsRaised, hasSize(0));
-        assertEquals(23, httpMessagesSent.size());
+        assertEquals(25, httpMessagesSent.size());
     }
 
-    @Test
-    void shouldAlertIfOkObtained() throws Exception {
+    @ParameterizedTest
+    @MethodSource("createTestPatterns")
+    void shouldAlertIfOkObtained(String patternName, String patternValue) throws Exception {
         // Given
-        nano.addHandler(new ForbiddenResponse(PROTECTED_PATH));
-        nano.addHandler(new OkResponse("/." + PROTECTED_PATH)); // Period is %2e
-        HttpMessage msg = this.getHttpMessage(PROTECTED_PATH);
+        nano.addHandler(new OkResponse("/" + patternName + PROTECTED_PATH + patternValue));
+        nano.addHandler(new ForbiddenResponse("/" + patternName + PROTECTED_PATH));
+        HttpMessage msg = this.getHttpMessage("/" + patternName + PROTECTED_PATH);
         rule.init(msg, this.parent);
         // When
         rule.scan();
         // Then
         assertThat(alertsRaised, hasSize(1));
         Alert alert = alertsRaised.get(0);
-        assertEquals("/%2e" + PROTECTED_PATH, alert.getAttack());
```

Comment:
Should be restored.

Reply:
I've got this restore for 6/7 of the tests. But I need help for testing the slash payload because nano handles `/./` the same as `/` so things actually alert earlier than expected.

Reply:
Resolved, it was the order I was adding the handler(s) in, conditionally added another handler.